ngraph/frontend/onnx_import/src/op/constant.cpp

```diff
@@ -165,6 +167,171 @@ namespace ngraph
                     default: throw error::tensor::invalid_data_type{tensor};
                     }
                 }
+
+                template <typename T>
+                std::vector<T> get_dense_vector(const std::vector<T>& values,
+                                                const std::vector<int64_t>& indices,
+                                                const size_t size)
+                {
+                    std::vector<T> dense_values(size);
+                    for (size_t i = 0; i < values.size(); ++i)
+                    {
+                        dense_values.at(indices.at(i)) = values.at(i);
+                    }
+                    return dense_values;
+                }
+
+                std::shared_ptr<default_opset::Constant>
+                    get_dense_tensor_as_constant(const std::vector<int64_t>& absolute_indices,
+                                                 const Tensor& values_tensor,
+                                                 const Shape& shape)
+                {
+                    size_t all_elements_number = 1;
+                    for (auto dim : shape)
+                        all_elements_number *= dim;
+
+                    switch (values_tensor.get_ng_type())
+                    {
+                    case element::boolean:
+                    {
+                        auto values = values_tensor.get_data<char>();
```

Comment:
Can you create a template function and use it in every `case`? There's a lot of duplication in here.

Reply:
Done